python/cuml/utils/numba_utils.py

```diff
@@ -139,3 +140,46 @@ def zeros(size, dtype, order='F'):
         gpu_zeros_1d.forall(size)(out)
 
     return out
+
+
+def device_array_from_ptr(ptr, shape, dtype, order='F', stride=None):
+    """Create a DeviceNDArray from a device pointer
+
+    Similar to numba.cuda.from_cuda_array_interface, difference is that we
+    allow Fortran array order here.
+
+    Parameters
+    ----------
+    ptr : int
+        device pointer
+    shape : tuple of ints
+    dtype : type of the data
+    order : 'C' or 'F'
+    stride : tuple of ints
+        Stride in bytes along each dimension the array. If it is left empty,
+        then will be filled automatically using the order parameter
+    """
+    dtype = np.dtype(dtype)
+    itemsize = dtype.itemsize
+    if stride is None:
+        stride = stride_from_order(shape, order, itemsize)
+    size = cuda.driver.memory_size_from_info(shape, stride, itemsize)
+    devptr = cuda.driver.get_devptr_for_active_ctx(ptr)
+    data = cuda.driver.MemoryPointer(cuda.current_context(),
+                                     devptr, size=size, owner=None)
+    device_array = cuda.devicearray.DeviceNDArray(
```

Comment:
This should not be using numba.cuda but RMM instead: https://github.com/rapidsai/rmm/blob/eb9305f0d4fc13286a9c6978f46beb024b6e9226/python/librmm_cffi/wrapper.py#L197 and also seems like a function that should eventually live in RMM so opening a github issue would be a good idea

Reply:
The current implementation of device_array_from_ptr in RMM works with 1D arrays only, otherwise they seem to use the same numba.cuda classes underneath. But you are right that this should be part of RMM, I have created Issue https://github.com/rapidsai/rmm/issues/135

Reply:
@dantegd - is it ok to resolve this convo? The issue is filed so I think we can move ahead.

Reply:
Resolving based on Slack comment.